### PR TITLE
pantheon-terminal: fixes "Using the 'memory' GSettings backend" issue

### LINKED ISSUE #10171: pantheon-terminal: window-size not saved
```
$ pantheon-terminal 
[_LOG_LEVEL_INFO 22:05:38.383910] Application.vala:153: Terminal version: 0.3.1.3
[_LOG_LEVEL_INFO 22:05:38.383989] Application.vala:155: Kernel version: 3.18.21
[_LOG_LEVEL_INFO 22:05:38.387631] [GLib-GIO] Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
[_LOG_LEVEL_WARN 22:05:38.387856] [GLib-GObject] The property GSettings:schema is deprecated and shouldn't be used anymore. It will be removed in a future version.
[_LOG_LEVEL_DEBUG 22:05:38.387923] Settings.vala:158: Loading settings from schema 'org.pantheon.terminal.saved-state'
[_LOG_LEVEL_DEBUG 22:05:38.388443] Settings.vala:158: Loading settings from schema 'org.pantheon.terminal.settings'
[_LOG_LEVEL_WARN 22:05:38.709549] Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
```

cc @ollieh

### PR DESCRIPTION
**Pull request by davidak**
###### Motivation for this change

fix annoying issue

/cc @jtojnar

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/pantheon/apps/pantheon-terminal/default.nix [hunk 1]
```diff
@@ -16,7 +16,8 @@ stdenv.mkDerivation rec {
   preFixup = ''
     for f in $out/bin/*; do
       wrapProgram $f \
-        --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:$XDG_ICON_DIRS:$out/share"
```
Comment (jtojnar) [inline pkgs/desktops/pantheon/apps/pantheon-terminal/default.nix]: Could you use `wrapGAppsHook` instead?

**Comment by jtojnar** [comment]
Also the `preConfigure` attribute is not necessary, it points to a wrong path anyway.

File: pkgs/desktops/pantheon/apps/pantheon-terminal/default.nix (hunk 1 above)
Comment (davidak) [reply pkgs/desktops/pantheon/apps/pantheon-terminal/default.nix]: sure

Comment (davidak) [reply pkgs/desktops/pantheon/apps/pantheon-terminal/default.nix]: like this?

File: pkgs/desktops/pantheon/apps/pantheon-terminal/default.nix [hunk 2]
```diff
@@ -9,25 +9,14 @@ stdenv.mkDerivation rec {
...
   nativeBuildInputs = [
-    perl cmake vala_0_38 pkgconfig makeWrapper
+    perl cmake vala_0_38 pkgconfig
     # For setup hook
     gobjectIntrospection
   ];
   buildInputs = with gnome3; [
     glib gtk3 granite libnotify gettext vte_290 libgee
-    gsettings-desktop-schemas defaultIconTheme
+    gsettings-desktop-schemas defaultIconTheme wrapGAppsHook
```
[the remark this answers is not in the document]
Comment (davidak) [reply pkgs/desktops/pantheon/apps/pantheon-terminal/default.nix]: changed and tested.

**Comment by jtojnar** [comment]
@GrahamcOfBorg build pantheon.pantheon-terminal

**Pull request merged**